relay-server/src/actors/processor.rs

```diff
@@ -1756,9 +1757,19 @@ impl EnvelopeProcessorService {
         // event id. To be defensive, we always overwrite to ensure consistency.
         event.id = Annotated::new(event_id);
 
-        // In processing mode, also write metrics into the event. Most metrics have already been
-        // collected at this state, except for the combined size of all attachments.
         if self.config.processing_enabled() {
+            // add the replay_id to the event's context if it exists on the DSC
+            if let Some(dsc) = envelope.dsc() {
```

Comment:
i wasn't sure where to stick this bit of logic but added here for now in `finalize_event`